client/app/telecom/pack/xdsl/access/comfortExchange/pack-xdsl-access-comfort-exchange.controller.js

```diff
@@ -0,0 +1,57 @@
+import { COMFORT_EXCHANGE_TYPE_ERROR } from './pack-xdsl-access-comfort-exchange.constant';
+
+angular.module('managerApp').controller('XdslAccessComfortExchangeCtrl', class XdslAccessComfortExchangeCtrl {
+  /* @ngInject */
+  constructor($stateParams, $translate, OvhApiXdsl, TucToast, TucToastError) {
+    this.$stateParams = $stateParams;
+    this.$translate = $translate;
+    this.OvhApiXdsl = OvhApiXdsl;
+    this.TucToast = TucToast;
+    this.TucToastError = TucToastError;
+  }
+
+  /*= =====================================
+  =            INITIALIZATION            =
+  ====================================== */
+  $onInit() {
+    this.exchange = {
+      order: {
+        url: null,
+        id: null,
+        priceWithoutTax: null,
+        priceWithTax: null,
+        tax: null,
+      },
+      isSuccess: false,
+    };
+    this.isAvailable = false;
+  }
+
+
+  comfortExchange() {
+    return this.OvhApiXdsl.Modem().v6().comfortExchange({
+      xdslId: this.$stateParams.serviceName,
```

Comment:
This could be injected through the state's resolve and avoid creating dependency to `$stateParams` in your controller